### DISCUSSION TITLE
Using tauri as backend server ??

### DISCUSSION THREAD

**Discussion by erikpa1** [tauri-apps/tauri#7857] [Ideas]
Hi, I think this is not possible for now (or at least the best idea, and not so easy),

but theoretically is it possible to take tauri, put it on some docker/virtual machine and run it as backend for my frontend app ?

Why I'm asking ? - Because I want to achieve total fullstack with tauri, because now I'm  switching inside my app like:
```
if (App.IsDesktop()) {
  invoke...
} else {
  fetch()
}
```

instead I could only do only invoke or fetch for example... :)

**Comment by FabianLars** [comment]
Tauri is defenitely not designed to be some kind of server and any of the assets and invoke calls can not be accessed from anywhere outside of the webview but there are these plugins: https://github.com/tauri-apps/plugins-workspace/tree/v1/plugins/localhost (for your frontend files) / https://github.com/tauri-apps/tauri-invoke-http/ (for invoke) that can expose it on localhost from where you could forward it to the outside world (don't take this as a recommendation, tauri doesn't do that by default for a reason).

Note however that you still need to ship the app to users, where they wouldn't _need_ to use the remote files unless of course you want them all to open the frontend in a browser at what point a dedicated server would for sure be a better choice.

**Comment by erikpa1** [comment]
Ok, thanks for your response :) Anyway I'm writing backend in actix-web, just had this Idea, to cover whole stack development with one platform :) Have a nice day.
